micro-rdk/src/common/system.rs

```diff
@@ -106,37 +130,88 @@ pub(crate) async fn force_shutdown(app_client: Option<AppClient>) {
     }
     match *SHUTDOWN_EVENT.lock().await {
         Some(SystemEvent::Restart) => terminate(),
-        Some(SystemEvent::DeepSleep(dur)) => {
+        Some(SystemEvent::DeepSleep {
+            duration,
+            ulp_enabled,
+        }) => {
             #[cfg(feature = "esp32")]
             {
-                if let Some(dur) = dur {
+                let mut result: sys::esp_err_t;
+
+                // disable other wakeup sources before setting
+                unsafe {
+                    result = sys::esp_sleep_disable_wakeup_source(
```

Comment:
let's remove this for now. We don't support all wakeup source so a module might make the decision to set up it's own wakeup sources for some reasons